src/server.c

```diff
@@ -5164,19 +5791,91 @@ NULL
         addReplyArrayLen(c, dictSize(server.commands));
         di = dictGetIterator(server.commands);
         while ((de = dictNext(di)) != NULL) {
-            addReplyCommand(c, dictGetVal(de));
+            addReplyCommandLegacy(c, dictGetVal(de));
         }
         dictReleaseIterator(di);
     } else if (!strcasecmp(c->argv[1]->ptr, "info")) {
         int i;
         addReplyArrayLen(c, c->argc-2);
         for (i = 2; i < c->argc; i++) {
-            addReplyCommand(c, dictFetchValue(server.commands, c->argv[i]->ptr));
+            addReplyCommandLegacy(c, lookupCommandBySds(c->argv[i]->ptr));
+        }
+    } else if (!strcasecmp(c->argv[1]->ptr, "count") && c->argc == 2) {
+        addReplyLongLong(c, dictSize(server.commands));
+    } else if (!strcasecmp(c->argv[1]->ptr,"getkeys") && c->argc >= 3) {
+        struct redisCommand *cmd = lookupCommand(c->argv+2,c->argc-2);
+        getKeysResult result = GETKEYS_RESULT_INIT;
+        int j;
+
+        if (!cmd) {
+            addReplyError(c,"Invalid command specified");
+            return;
+        } else if (cmd->getkeys_proc == NULL && cmd->key_specs_num == 0) {
+            addReplyError(c,"The command has no key arguments");
+            return;
+        } else if ((cmd->arity > 0 && cmd->arity != c->argc-2) ||
+                   ((c->argc-2) < -cmd->arity))
+        {
+            addReplyError(c,"Invalid number of arguments specified for command");
+            return;
+        }
+
+        if (!getKeysFromCommand(cmd,c->argv+2,c->argc-2,&result)) {
+            addReplyError(c,"Invalid arguments specified for command");
+        } else {
+            addReplyArrayLen(c,result.numkeys);
+            for (j = 0; j < result.numkeys; j++) addReplyBulk(c,c->argv[result.keys[j]+2]);
+        }
+        getKeysFreeResult(&result);
+    } else {
+        addReplySubcommandSyntaxError(c);
+    }
+}
```

Comment:
have a clue why did git decide all these lines are new / changed?

Reply:
yes, because COMMANDS is directly after COMMAND and they have the same start and end

do you want me to move it?

Reply:
ohh
it's odd to see the old code being marked as new, and the new code being marked as old.
maybe if we extract the GETKEYS implementation it'll change all that.